source/utils.js

```diff
@@ -37,27 +37,59 @@ const convertBlobToArrayBufferView = blob =>
         fileReader.readAsArrayBuffer(blob);
     });
 
-
 const mergeUint8Arrays = (arr1, arr2) => {
     const merge = new Uint8Array(arr1.length + arr2.length);
     merge.set(arr1);
     merge.set(arr2, arr1.length);
     return merge;
 };
 
+/**
+ * converts an ArrayBuffer to a hexadecimal string
+ *
+ * @param {ArrayBuffer} buffer
+ * @returns {String} - hexadecimal representation of the ArrayBuffer
+ *
+ */
+const convertArrayBufferToHexadecimal = buffer => Array.from(new Uint8Array(buffer))
+    .map(x => x.toString(16).padStart(2, '0'))
+    .join('');
+
+
+/**
+ * converts a hexadecimal string to an ArrayBuffer
+ *
+ * @param {String} hexadecimal
+ * @returns {Uint8Array} - the bytes of the hexadecimal string
+ *
+ */
+const convertHexadecimalToArrayBuffer = (hexadecimal) => {
+    if (typeof hexadecimal !== 'string') {
```

Comment:
I see that you added a test for the happy case. Could you also add a test for these errors?

Reply:
Update with 8dff068